nn/transformer.py

```diff
@@ -0,0 +1,220 @@
+"""
+Transformer Modules
+"""
+
+from __future__ import annotations
+from .base import Module, LayerRef
+from .math_ import relu
+from ._generated_layers import Linear, layer_norm
+from .utils import dropout
+from .container import ModuleList
+from typing import Optional, Any
+
+
+class TransformerEncoderLayer(Module):
+  """
+  Defines one layer of a standard transformer encoder
+  """
+  def __init__(self, d_model, nhead, dim_feedforward=2048, drop=0.1, act=relu, layer_norm_eps=1e-5,
```

Comment:
Types should be specified.

The param names are a bit inconsistent to each other. `d` vs `dim`. `nhead` vs `n_head` vs `num_head`.

Reply:
I updated a bunch of typing where I deemed it reasonable. 
One question regarding typing: For the activation functions, I put `activation` as a type since in the end `relu` etc. just "wrap" `activation`. Does that work for automatic typing checks or will that confuse the system?